File: backend/app/schemas/review.py
"""Review workflow schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ReviewTask(BaseModel):
    id: str
    document_id: str
    assigned_to: Optional[str] = None
    status: str  # pending | in_progress | approved | rejected | escalated
    priority: str
    notes: Optional[str] = None
    created_at: datetime
    resolved_at: Optional[datetime] = None


class ReviewAction(BaseModel):
    action: str  # approve | reject | escalate
    notes: Optional[str] = None

File: frontend/src/types/review.ts
// Review types
export interface ReviewTask {
  id: string;
  document_id: string;
  assigned_to: string | null;
  status: 'pending' | 'in_progress' | 'approved' | 'rejected' | 'escalated';
  priority: string;
  notes: string | null;
  created_at: string;
  resolved_at: string | null;
}
